### PR TITLE
Allow to accept bookings in slots before a booking

### PR DESCRIPTION
**Pull request by sewa**

### CODE REVIEW

File: spec/trixer/slotter_spec.rb [hunk 1]
```diff
@@ -173,19 +173,32 @@
...
+        it { expect(subject[60]).to eql(capacity: 2, free_duration: { rest: 3 }) }
+        it { expect(subject[61]).to eql(capacity: 2, free_duration: { rest: 2 }) }
+        it { expect(subject[62]).to eql(capacity: 2, free_duration: { rest: 1 }) }
         it { expect(subject[64]).to eql(capacity: 2, booking: 1, free_duration: { rest: 0, 1 => 6 }) }
         it { expect(subject[65]).to eql(capacity: 2, booking: 1, free_duration: { rest: 0, 1 => 5 }) }
         it { expect(subject[66]).to eql(capacity: 2, booking: 1, free_duration: { rest: 0, 1 => 4 }) }
         it { expect(subject[67]).to eql(capacity: 2, booking: 1, free_duration: { rest: 0, 1 => 3 }) }
-        it { expect(subject[69]).to eql(capacity: 2, free_duration: { rest: 1 }) }
+        it { expect(subject[68]).to eql(capacity: 2, free_duration: { rest: 2, 2 => 5 }) }
+        it { expect(subject[69]).to eql(capacity: 2, free_duration: { rest: 1, 2 => 5 }) }
```
Comment (larskuhnt) [inline spec/trixer/slotter_spec.rb:184]: it should increment for the current booking, so the expected value here would be `{ rest: 1, 2 => 6 }`

File: lib/trixer/slotter.rb [hunk 2]
```diff
@@ -192,11 +192,16 @@ def place_slot_data(place_id:)
...
           slot_data[:free_duration] = { rest: 0, booking.id => current_booking_length - 1 }
+          current_booking = nil if @gap_slots[slot] == true
         elsif @gap_slots[slot] == true
           leng = current_length
           current_length = 1
           current_booking_length = 1
+          current_booking = nil
           slot_data[:free_duration] = { rest: leng }
+        elsif !booking && current_booking
+          current_length += 1
```
Comment (larskuhnt) [inline lib/trixer/slotter.rb:203]: I think you need to increment `current_booking_length` here as well

File: spec/trixer/slotter_spec.rb (hunk 1 above)
Comment (sewa) [reply spec/trixer/slotter_spec.rb:184]: at that point the max possible length is reached and to continue counting would mean to add code that is not really required.

File: lib/trixer/slotter.rb (hunk 2 above)
Comment (sewa) [reply lib/trixer/slotter.rb:203]: see comment above

File: spec/trixer/slotter_spec.rb (hunk 1 above)
Comment (larskuhnt) [reply spec/trixer/slotter_spec.rb:184]: what if we want to add the feature to extend a booking to the left (start earlier). that would work only for one slot.

**Pull request merged**